File: dsa-learning/dsa-java-code/src/main/java/com/dsa/leetcode/labuladong2/array/P27_RemoveElement.java
package com.dsa.leetcode.labuladong2.array;

public class P27_RemoveElement {

    public static int removeElement(int[] nums, int val) {
        if (nums == null || nums.length == 0) {
            return 0;
        }
        int slow = 0;
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] != val) {
                nums[slow++] = nums[i];
            }
        }
        return slow;
    }

    public static void main(String[] args) {
        int[] nums = { 3, 2, 2, 3 };
        int size = removeElement(nums, 3);
        System.out.println(size);
    }

}

File: dsa-learning/dsa-python-code/com/dsa/leetcode/labuladong2/array/P27_RemoveElement.py
from typing import List


class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        length = len(nums)
        if length == 0:
            return 0
        slow = 0
        for i in range(length):
            if nums[i] != val:
                nums[slow] = nums[i]
                slow = slow + 1
        return slow